src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.targets

```diff
@@ -223,7 +223,10 @@ Copyright (c) .NET Foundation. All rights reserved.
   <PropertyGroup Condition="'$(MSBuildRuntimeType)' == 'Full'">
     <!-- Automatically opt users into using the toolset package if they are running an MSBuild other than what this SDK was built against.
           This is to reduce 'tearing'/dependency mismatch, but as always users can override this behavior by disabling the hosted compiler flag. -->
-    <BuildWithNetFrameworkHostedCompiler Condition="'$(BuildWithNetFrameworkHostedCompiler)' == '' and '$(_IsDisjointMSBuildVersion)' == 'true'">true</BuildWithNetFrameworkHostedCompiler>
+    <BuildWithNetFrameworkHostedCompiler Condition="'$(BuildWithNetFrameworkHostedCompiler)' == ''
+                                                    and '$(_IsDisjointMSBuildVersion)' == 'true'
+                                                    and ('$(Language)' == 'C#'
+                                                    or '$(Language)' == 'VB')">true</BuildWithNetFrameworkHostedCompiler>
```

Comment:
Are we confident this is a place where we are guaranteed to have `$(Language)` set correctly? Know in the past got bit a few times because it wasn't guaranteed to be set.

Reply:
yes - the language is set in the [language-specific targets](https://github.com/dotnet/sdk/blob/3efd0d23f755cfe02f901f3d5f72665ca6987997/src/Tasks/Microsoft.NET.Build.Tasks/sdk/Sdk.targets#L24-L37) when they are imported, then this file is imported [almost immediately after](https://github.com/dotnet/sdk/blob/3efd0d23f755cfe02f901f3d5f72665ca6987997/src/Tasks/Microsoft.NET.Build.Tasks/sdk/Sdk.targets#L39-L42).